### PR TITLE
Ability to assume a specific role

### PR DESCRIPTION
**Pull request by marcin-caban**
Plus a bug fix in S3Download (to use the defined client).

### CODE REVIEW

**Comment by marcin-caban** [comment]
Apparently travis configuration is broken, missing AWS credentials.

I've built and tested it locally with role set, it works.

Could you, please, release a new version with my change? 

Thanks in advance.

Cheers
Marcin

**Comment by marcin-caban** [comment]
@mgk Could you have a look on this?

**Comment by mgk** [comment]
Sorry for delay. Looking at it now.

File: src/main/groovy/com/github/mgk/gradle/S3Plugin.groovy [hunk 1]
```diff
@@ -33,9 +34,14 @@ abstract class S3Task extends DefaultTask {
     @Input
     String bucket
 
+    static AmazonS3Client s3Client
+
     String getBucket() { bucket ?: project.s3.bucket }
 
     def getS3Client() {
+        if (s3Client) {
```
Comment (mgk) [inline src/main/groovy/com/github/mgk/gradle/S3Plugin.groovy:42]: I have run into issues reusing the `s3Client` object (I seem to recall it expiring or becoming invalid after a certain period of time) so I create it for each operation.

If you really want s3Client reuse, please put that in a separate PR and make it configurable via an environment variable.

**Review by mgk — CHANGES_REQUESTED** [review]
LGTM otherwise. Build succeeds for me locally so I expect once merged the Travis build will succeed also.

**Comment by mgk** [comment]
Appears abandoned. Closing. Please reopen if needed (and not feedback above). Thanks.

**Pull request closed without merging**